src/lib/deck.rs

```diff
@@ -1,16 +1,46 @@
 use crate::lib::card::*;
-use rand::seq::SliceRandom;
-use rand::Rng;
+use std::collections::VecDeque;
 
-pub struct Deck;
+pub struct Deck {
+    cards: VecDeque<Card>,
+}
 
 impl Deck {
-    pub fn draw_card(&self) -> Card {
-        let random_value: u8 = rand::thread_rng().gen_range(1..14);
-        let mut rng = rand::thread_rng();
-        let suits = Suit::all();
-        let suit = suits.choose(&mut rng).unwrap();
+    pub fn new() -> Self {
+        let mut cards = VecDeque::new();
+        for value in 1..=13 {
+            for suit in Suit::all() {
+                cards.push_back(Card::new(value, suit))
+            }
+        }
+        Self { cards }
+    }
+
+    fn from_cards(cards: &[Card]) -> Self {
+        let cards = VecDeque::from(cards.to_vec());
```

Comment:
Add `cargo clippy` to your build process. I believe `VecDeque::from(&cards)` is sufficient